core/modules/qdisp/QueryResource.h

```diff
@@ -56,17 +56,23 @@ class QueryResource : public XrdSsiService::Resource {
                   boost::shared_ptr<util::UnaryCallable<void, bool> > finishFunc,
                   boost::shared_ptr<util::VoidCallable<void> > retryFunc,
                   ExecStatus& status)
-        : Resource(rPath.c_str()),
+        : Resource(::strdup(rPath.c_str())), // this char* must live as long as this object, so copy it on heap
           _session(NULL),
           _request(NULL),
           _payload(payload),
           _requester(requester),
           _finishFunc(finishFunc),
           _retryFunc(retryFunc),
-          _status(status) {
+          _status(status)
+    {
+        if (rName == NULL) {
+            throw std::bad_alloc();
+        }
     }
 
-    virtual ~QueryResource() {}
+    virtual ~QueryResource() {
+        free(const_cast<char *>(rName)); // clean up heap allocated resource path copy
```

Comment:
nitpick: `std::free` or `::free` (depending on whether you included `<cstdlib>` or `<stdlib.h>`)